turbopack/crates/turbopack/src/lib.rs

```diff
@@ -697,32 +728,104 @@ impl AssetContext for ModuleAssetContext {
     ) -> Result<Vc<ModuleResolveResult>> {
         let this = self.await?;
 
-        let result = result
+        let replace_externals = this.replace_externals;
+        let import_externals = this
+            .module_options_context
             .await?
-            .map_module(|source| {
+            .ecmascript
+            .import_externals;
+
+        let result = result.await?;
+
+        let affecting_sources = &result.affecting_sources;
+
+        let result = result
+            .map_items_module(|item| {
                 let reference_type = reference_type.clone();
                 async move {
-                    let process_result = self.process(*source, reference_type);
-                    Ok(match *process_result.await? {
-                        ProcessResult::Module(m) => {
-                            ModuleResolveResultItem::Module(ResolvedVc::upcast(m))
+                    Ok(match item {
+                        ResolveResultItem::Source(source) => {
+                            match &*self.process(*source, reference_type).await? {
+                                ProcessResult::Module(module) => {
+                                    ModuleResolveResultItem::Module(*module)
+                                }
+                                ProcessResult::Unknown(source) => {
+                                    ModuleResolveResultItem::Unknown(*source)
+                                }
+                                ProcessResult::Ignore => ModuleResolveResultItem::Ignore,
+                            }
+                        }
+                        ResolveResultItem::External { name, ty, traced } => {
+                            let replacement = if replace_externals {
+                                let additional_refs: Vec<Vc<Box<dyn ModuleReference>>> = if let (
+                                    ExternalTraced::Traced,
+                                    Some(tracing_root),
+                                ) = (
+                                    traced,
+                                    self.module_options_context()
+                                        .await?
+                                        .enable_externals_tracing,
+                                ) {
+                                    let externals_context = externals_tracing_module_context(ty);
+                                    let out_dir = tracing_root.join("index".into());
```

Comment:
We used `_` before when we don't have a filename

```suggestion
                                    let out_dir = tracing_root.join("_".into());
```